core/metadata_test.cc

```diff
@@ -0,0 +1,220 @@
+#include "metadata.h"
+#include "module.h"
+
+#include <stdlib.h>
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+// Mocking out misc things  ------------------------------------------------
+
+// -------------------------------------------------------------------------
+
+class Foo : public Module {
```

Comment:
I think the mock code should be not exposed to the outside world, to avoid namespace pollution.  The Google C++ style guide says all code (except #include) in *.cc should be in a namespace. Please use an unnamed namespace here.